libraries/src/CMS/Component/ComponentHelper.php

```diff
@@ -355,6 +358,11 @@ public static function renderComponent($option, $params = array())
 			define('JPATH_COMPONENT_ADMINISTRATOR', JPATH_ADMINISTRATOR . '/components/' . $option);
 		}
 
+		if (!defined('JPATH_COMPONENT_API'))
```

Comment:
We're trying to get rid of these constants. Because they stop you running components from other components (i.e. when com_config calls com_template or something)